beacon-chain/execution/engine_client.go

```diff
@@ -511,6 +525,64 @@ func (s *Service) ReconstructFullBellatrixBlockBatch(
 	return unb, nil
 }
 
+// ReconstructBlobSidecars reconstructs the blob sidecars for a given beacon block.
+// It retrieves the KZG commitments from the block body, fetches the associated blobs and proofs,
+// and constructs the corresponding verified read-only blob sidecars.
+func (s *Service) ReconstructBlobSidecars(ctx context.Context, block interfaces.ReadOnlySignedBeaconBlock, blockRoot [32]byte) ([]blocks.VerifiedROBlob, error) {
+	blockBody := block.Block().Body()
+
+	// Get KZG commitments from the block body
+	kzgCommitments, err := blockBody.BlobKzgCommitments()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get blob KZG commitments")
+	}
+
+	// Initialize KZG hashes and retrieve blobs
+	kzgHashes := make([]common.Hash, len(kzgCommitments))
+	blobs, err := s.GetBlobs(ctx, kzgHashes)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get blobs")
+	}
+
+	// Get the block header and its hash tree root
+	header, err := block.Header()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get header")
+	}
+	// Loop through the blobs and reconstruct blob sidecars
+	verifiedBlobs := make([]blocks.VerifiedROBlob, 0, len(blobs))
+	for index, blob := range blobs {
+		if blob == nil {
+			continue
+		}
+
+		// Get the Merkle proof for the KZG commitment
+		proof, err := blocks.MerkleProofKZGCommitment(blockBody, index)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not get Merkle proof KZG commitment")
+		}
+
+		// Create the BlobSidecar object
+		sidecar := &ethpb.BlobSidecar{
+			Index:                    uint64(index),
+			Blob:                     blob.Blob,
+			KzgCommitment:            kzgCommitments[index],
+			KzgProof:                 blob.KzgProof,
+			SignedBlockHeader:        header,
+			CommitmentInclusionProof: proof,
+		}
+
+		// Create a read-only blob with the header root
+		roBlob, err := blocks.NewROBlobWithRoot(sidecar, blockRoot)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not create RO blob with root")
+		}
+		verifiedBlobs = append(verifiedBlobs, blocks.NewVerifiedROBlob(roBlob))
```

Comment:
You're calling NewVerifiedROBlob and minting VerifiedROBlobs here, rather than getting those values from a verifier. That is risky. Would it be possible to define a verifier for this? Maybe something like the initial sync verifier that can deal with a batch. 

On that note, I don't see you verifying the commitment inclusion proof anywhere in this PR.